pkg/handler/collector/s3/bucket/encoding.go

```diff
@@ -0,0 +1,52 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bucket
+
+import (
+	"strings"
+
+	"github.com/guacsec/guac/pkg/handler/processor"
+)
+
+const (
+	BZIP2   = "BZIP2"
+	ZSTD    = "ZSTD"
+	UNKNOWN = "UNKNOWN"
+)
+
+func ExtractEncoding(encoding string, filename string) processor.EncodingType {
+	switch encoding {
+	case BZIP2:
+		return BZIP2
```

Comment:
use the enums in the [processor](https://github.com/guacsec/guac/blob/main/pkg/handler/processor/processor.go#L86-L87), and if not present, add them to the file